src/runtime-helpers/maker.ts

```diff
@@ -47,48 +61,50 @@ module Experimental {
-
-                //                 if (['business', 'personal'].indexOf(driveType) < 0) {
-                //                     throw new Error('ScriptLab can only currently find urls for files stored on OneDrive consumer and business.');
-                //                 }
-
-                //                 let sliceIndex = driveType === 'business' ? 4 : 3;
-                //                 let path = tempUrl.split('/').slice(sliceIndex).join('/');
-                //                 let documentUrl = `${graphBaseUrl}/me/drive/root:/${path}:/workbook`;
-
-                //                 worker.postMessage([accessToken, documentUrl, makerCode.toString()]);
-                //             } else {
-                //                 throw new Error('Could not retrieve driveType.');
-                //             }
-                //         };
-
-                //         xhr.send();
-
-                //     } else {
-                //         throw new Error('You must specify a documentUrl if you are not inside of Excel!');
-                //     }
-                // }
+                let activeDocumentUrl = await getActiveDocumentUrl(accessToken);
+
+                worker.postMessage({
+                    accessToken,
+                    makerCode: makerCode.toString(),
+                    activeDocumentUrl,
+                    scriptReferences: _scriptReferences
+                });
+            });
+        }
+
+        async function getActiveDocumentUrl(accessToken: string) {
+            return new Promise((resolve, reject) => {
+                if (window && (window as any).Office && (window as any).Office.context) {
+                    // get document url
+                    let tempUrl = Office.context.document.url.replace('https://', '');
+                    let graphBaseUrl = 'https://graph.microsoft.com/v1.0';
+
+                    let xhr = new XMLHttpRequest();
+                    xhr.open('GET', `${graphBaseUrl}/me/drive`);
+                    xhr.setRequestHeader('Authorization', `Bearer ${accessToken}`);
+                    xhr.setRequestHeader('content-type', 'application/json');
+                    xhr.onload = () => {
+                        if (xhr.readyState === 4 && xhr.status === 200) {
+                            let driveType = JSON.parse(xhr.responseText).driveType;
+
+                            if (['business', 'personal'].indexOf(driveType) < 0) {
+                                throw new Error('Currently, ScriptLab can only find urls for files stored on OneDrive consumer and business.');
+                            }
+
+                            let sliceIndex = driveType === 'business' ? 4 : 3;
+                            let path = tempUrl.split('/').slice(sliceIndex).join('/');
+                            let documentUrl = `${graphBaseUrl}/me/drive/root:/${path}:/workbook`;
+
+                            resolve(documentUrl);
+                        } else {
```

Comment:
Is there also an onfail sort of method on xhr, or does it always fire "onload" even on catostrophic error?